### DISCUSSION TITLE
How to close file properly in nvim-tree

### DISCUSSION THREAD

**Discussion by bavo96** [nvim-tree/nvim-tree.lua#2102] [General]
Using the default settings of nvim-tree, when I have a lot of opened files and I want to close a single file, nvim-tree closes the buffer and only the tree appears.

https://user-images.githubusercontent.com/27908949/229303630-1a761742-6209-4ecb-a337-63c14f2e132a.mp4

After I use the auto-close code reference in https://github.com/nvim-tree/nvim-tree.lua/wiki/Auto-Close#eliasa5, when I close the file, it closes my neovim instead of a single file.

https://user-images.githubusercontent.com/27908949/229303792-2f0c059d-a4de-452b-8b2c-af340218ce16.mp4

It also affects the packer.nvim when I press `:PackerSync`. The tree closes after I hit the command and pops up a bunch of error, which has a message `Invalid window id: 1002`.

https://user-images.githubusercontent.com/27908949/229303958-b987e1a6-2431-40dc-9dee-eaf5b8a3a115.mp4

Is this a normal behavior and I should accept that? Please help :(

**Comment by bavo96** [comment]
I found out the solution https://github.com/nvim-tree/nvim-tree.lua/wiki/Auto-Close#beauwilliams fix my 3rd clip, but still close neovim like 2nd clip when I hit `:q`. I'm using `map('n', '<A-c>', '<Cmd>BufferClose<CR>', opts)`  in `barbar.nvim` to close a buffer instead of `:q` and `:q` to quit the whole neovim, not sure if this is a correct way :D

**Comment by alex-courtis** [comment]
There is a lot going on here: it sounds like you have a lot of plugins and automation that could result in this behaviour.

Suggestion: start with a [Clean Room](https://github.com/nvim-tree/nvim-tree.lua/wiki/Troubleshooting#clean-room-replication) and add plugins / automation until you can reproduce the situation.
